server.js

```diff
@@ -210,10 +211,41 @@ module.exports.initExpress = function() {
         logger.info(`localRootFilesDir: Mapping ${config.localRootFilesDir} into /`);
         app.use(express.static(config.localRootFilesDir));
     }
+
+    // To allow for more aggressive caching of static files served from public/,
+    // we introduce a new `assets/` path that includes a cachebuster in the path.
+    // In requests for resources, the cachebuster will be a hash of the contents
+    // of `/public`, which we will compute at startup. See `lib/assets.js` for
+    // implementation details.
+    app.use((req, res, next) => {
+        res.locals.asset_path = assets.assetPath;
+        next();
+    });
+    app.use('/assets/:cachebuster', express.static(path.join(__dirname, 'public'), {
+        maxAge: '31557600',
```

Comment:
Should we also set `immutable=true` and `etags=false`? See http://expressjs.com/en/resources/middleware/serve-static.html

Reply:
I think we should definitely set `immutable=true`. For browsers that don't understand the `immutable` header, it miiiight still be a useful optimization to leave Etags enabled? At the very least, I don't think it'd hurt. For instance, at NerdWallet, we use `max-age=1year` but still leave Etags enabled (and don't even set `immutable`!).

Reply:
I agree that there doesn't seem to be much downside to leaving `etags` enabled.